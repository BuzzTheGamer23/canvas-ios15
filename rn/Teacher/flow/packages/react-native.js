// @flow
export type PanResponderGestureState = {
  stateID: number,
  moveX: number,
  moveY: number,
  x0: number,
  y0: number,
  dx: number,
  dy: number,
  vx: number,
  vy: number,
}
