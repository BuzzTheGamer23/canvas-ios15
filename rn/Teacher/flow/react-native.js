//
// Copyright (C) 2017-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow
/* eslint-disable */

// define basic style prop types
// the internal ones are the best we have
import {
  type ____ViewStyleProp_Internal,
  type ____TextStyleProp_Internal,
  type ____ImageStyleProp_Internal,
} from 'react-native/Libraries/StyleSheet/StyleSheetTypes'

declare type ViewStyleProp = ____ViewStyleProp_Internal
declare type TextStyleProp = ____TextStyleProp_Internal
declare type ImageStyleProp = ____ImageStyleProp_Internal
