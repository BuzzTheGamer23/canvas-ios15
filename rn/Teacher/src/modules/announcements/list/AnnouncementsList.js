//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

/* @flow */

import React, { Component } from 'react'
import { connect } from 'react-redux'
import {
  View,
  StyleSheet,
  FlatList,
} from 'react-native'
import i18n from 'format-message'
import moment from 'moment'

import Screen from '../../../routing/Screen'
import refresh from '../../../utils/refresh'
import Row from '../../../common/components/rows/Row'
import Actions from './actions'
import Images from '../../../images'

type State = AsyncState & {
  announcements: Discussion[],
  courseName: string,
}

type OwnProps = {
  courseID: string,
}

export type Props = OwnProps & State & typeof Actions & RefreshProps & NavigationProps

export class AnnouncementsList extends Component<any, Props, any> {

  render () {
    return (
      <Screen
        navBarStyle='dark'
        title={i18n('Announcements')}
        subtitle={this.props.courseName}
        rightBarButtons={[
          {
            image: Images.add,
            testID: 'announcements.list.addButton',
            action: this.addAnnouncement,
          },
        ]}
      >
        <View style={styles.container}>
          <FlatList
            data={this.props.announcements}
            renderItem={this.renderRow}
            keyExtractor={(item, index) => item.id}
            testID='announcements.list.list'
            refreshing={Boolean(this.props.pending)}
            onRefresh={this.props.refresh}
          />
        </View>
      </Screen>
    )
  }

  renderRow = ({ item, index }: { item: Discussion, index: number }) => {
    return (
      <Row
        title={item.title}
        subtitle={moment(item.delayed_post_at || item.posted_at).format(`MMM D [${i18n('at')}] h:mm A`)}
        border='bottom'
        height='auto'
        disclosureIndicator={true}
        testID={`announcements.list.announcement.row-${index}`}
        onPress={this.selectAnnouncement(item)}
      />
    )
  }

  addAnnouncement = () => {
    this.props.navigator.show(`/courses/${this.props.courseID}/announcements/new`, { modal: true })
  }

  selectAnnouncement (announcement: Discussion) {
    return () => this.props.navigator.show(announcement.html_url, { modal: false }, {
      isAnnouncement: true,
    })
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
})

export function mapStateToProps ({ entities }: AppState, { courseID }: OwnProps): State {
  let announcements = []
  let pending = 0
  let error = null
  let courseName = ''
  if (entities &&
    entities.courses &&
    entities.courses[courseID] &&
    entities.courses[courseID].announcements &&
    entities.discussions) {
    const course = entities.courses[courseID]
    const refs = course.announcements.refs
    pending = course.announcements.pending
    error = course.announcements.error
    if (course.course) {
      courseName = course.course.name
    }
    announcements = refs
      .map(ref => entities.discussions[ref].data)
  }
  return {
    announcements,
    courseName,
    pending,
    error,
  }
}

const Refreshed = refresh(
  props => {
    props.refreshAnnouncements(props.courseID)
  },
  props => props.announcements.length === 0,
  props => Boolean(props.pending)
)(AnnouncementsList)
const Connected = connect(mapStateToProps, Actions)(Refreshed)
export default (Connected: Component<any, Props, any>)
