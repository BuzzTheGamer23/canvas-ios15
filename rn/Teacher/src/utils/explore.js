/* @flow */

function hasProp (object: any, key: string, value: any): boolean {
  if (object instanceof Object) {
    const props = object.props
    if (props && props instanceof Object) {
      for (var prop in props) {
        if (prop === key && props[prop] === value) {
          return true
        }
      }
    }
  }

  return false
}

function selectByProp (object: any, propKey: string, value: any): ?any {
  if (!(typeof object === 'object')) {
    return null
  }

  if (hasProp(object, propKey, value)) {
    return object
  }

  let result = null

  for (var key in object) {
    if (object[key] instanceof Array) {
      object[key].forEach((e) => {
        result = selectByProp(e, propKey, value)
        if (result) {
          return result
        }
      })
    }

    result = selectByProp(object[key], propKey, value)
    if (result) {
      return result
    }
  }

  return result
}

class ComponentExplorer {
  component: any

  constructor (component: any) {
    this.component = component
  }

  selectByProp (propKey: string, value: any): ?any {
    return selectByProp(this.component, propKey, value)
  }

  selectByID (id: string): ?any {
    return this.selectByProp('testID', id)
  }
}

export default function explore (component: any): ComponentExplorer {
  return new ComponentExplorer(component)
}
