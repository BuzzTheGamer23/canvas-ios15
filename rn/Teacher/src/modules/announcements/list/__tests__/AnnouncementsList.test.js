//
// Copyright (C) 2016-present Instructure, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

/* @flow */

import React from 'react'
import 'react-native'
import renderer from 'react-test-renderer'

import { AnnouncementsList, type Props, mapStateToProps } from '../AnnouncementsList'
import explore from '../../../../../test/helpers/explore'

const template = {
  ...require('../../../../__templates__/discussion'),
  ...require('../../../../__templates__/helm'),
  ...require('../../../../redux/__templates__/app-state'),
}

jest
  .mock('Button', () => 'Button')
  .mock('TouchableHighlight', () => 'TouchableHighlight')
  .mock('TouchableOpacity', () => 'TouchableOpacity')
  .mock('../../../../routing/Screen')

describe('AnnouncementsList', () => {
  let props: Props
  beforeEach(() => {
    props = {
      courseID: '1',
      refreshing: false,
      refresh: jest.fn(),
      pending: 0,
      error: null,
      navigator: template.navigator(),
      announcements: [
        template.discussion({
          id: '1',
          title: 'Untitled',
          posted_at: '2013-11-14T16:55:00-07:00',
        }),
        template.discussion({
          id: '2',
          title: 'Getting Started in Biology 101',
          posted_at: '2013-10-28T14:16:00-07:00',
        }),
      ],
      refreshAnnouncements: jest.fn(),
      courseName: 'Im a course',
    }
  })

  it('renders', () => {
    testRender(props)
  })

  it('renders while pending', () => {
    props.announcements = []
    props.pending = 1
    testRender(props)
  })

  it('renders while refreshing', () => {
    props.pending = 1
    props.refreshing = true
    testRender(props)
  })

  it('renders empty list', () => {
    props.announcements = []
    testRender(props)
  })

  it('navigates to new announcement', () => {
    props.navigator.show = jest.fn()
    props.courseID = '2'
    tapAdd(render(props))
    expect(props.navigator.show).toHaveBeenCalledWith(
      '/courses/2/announcements/new',
      { modal: true },
    )
  })

  it('navigates to announcement when row tapped', () => {
    const announcement = template.discussion({ html_url: 'https://canvas.instructure.com/courses/1/discussions/2' })
    props.navigator.show = jest.fn()
    props.announcements = [announcement]
    const row: any = explore(render(props).toJSON()).selectByID('announcements.list.announcement.row-0')
    row.props.onPress()
    expect(props.navigator.show).toHaveBeenCalledWith(announcement.html_url, { modal: false }, {
      isAnnouncement: true,
    })
  })

  it('displays delayed post at date for delayed announcements', () => {
    const announcement = template.discussion({
      delayed_post_at: '3019-10-28T14:16:00-07:00',
      posted_at: '2017-10-27T14:16:00-07:00',
    })
    props.announcements = [announcement]
    const subtitle: any = explore(render(props).toJSON()).selectByID('announcements.list.announcement.row-0-subtitle-lbl')
    expect(subtitle.children).toEqual(['Oct 28 at 3:16 PM'])
  })

  function testRender (props: any) {
    expect(render(props)).toMatchSnapshot()
  }

  function render (props: any) {
    return renderer.create(<AnnouncementsList {...props} />)
  }

  function tapAdd (component: any) {
    const addBtn: any = explore(component.toJSON()).selectRightBarButton('announcements.list.addButton')
    addBtn.action()
  }
})

describe('mapStateToProps', () => {
  it('maps course announcement refs to props', () => {
    const one = template.discussion({ id: '1' })
    const two = template.discussion({ id: '2' })
    const three = template.discussion({ id: '3' })
    const state = template.appState({
      entities: {
        courses: {
          '1': {
            announcements: {
              pending: 1,
              error: null,
              refs: ['1', '3'],
            },
            course: {
              name: 'CS 1010',
            },
          },
        },
        discussions: {
          '1': {
            data: one,
          },
          '2': {
            data: two,
          },
          '3': {
            data: three,
          },
        },
      },
    })

    expect(
      mapStateToProps(state, { courseID: '1' })
    ).toEqual({
      announcements: [one, three],
      pending: 1,
      courseName: 'CS 1010',
      error: null,
    })
  })

  it('maps empty state', () => {
    const state = template.appState({
      entities: {},
    })
    expect(
      mapStateToProps(state, { courseID: '1' })
    ).toEqual({
      announcements: [],
      courseName: '',
      pending: 0,
      error: null,
    })
  })

  it('maps without course', () => {
    const one = template.discussion({ id: '1' })
    const two = template.discussion({ id: '2' })
    const three = template.discussion({ id: '3' })
    const state = template.appState({
      entities: {
        courses: {
          '1': {
            announcements: {
              pending: 1,
              error: null,
              refs: ['1', '3'],
            },
          },
        },
        discussions: {
          '1': {
            data: one,
          },
          '2': {
            data: two,
          },
          '3': {
            data: three,
          },
        },
      },
    })

    expect(
      mapStateToProps(state, { courseID: '1' })
    ).toEqual({
      announcements: [one, three],
      pending: 1,
      courseName: '',
      error: null,
    })
  })
})
