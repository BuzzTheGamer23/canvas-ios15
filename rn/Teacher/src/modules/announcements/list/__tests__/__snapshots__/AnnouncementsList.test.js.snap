// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnnouncementsList renders 1`] = `
<Screen
  navBarStyle="dark"
  rightBarButtons={
    Array [
      Object {
        "action": [Function],
        "image": 1,
        "testID": "announcements.list.addButton",
      },
    ]
  }
  subtitle="Im a course"
  title="Announcements"
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      data={
        Array [
          Object {
            "allow_rating": false,
            "assignment": Object {
              "assignment_group_id": 1376,
              "automatic_peer_reviews": false,
              "course_id": 987654321,
              "created_at": "2017-03-17T19:15:25Z",
              "description": "a course that can take any beginner to professional in a matter of microseconds.  Harness the power of your favorite languages like PHP and Cold Fusion.",
              "due_at": "2037-06-01T05:59:00Z",
              "grading_type": "points",
              "html_url": "http://hostname.tld/whatever",
              "id": 123456789,
              "lock_at": "2037-06-01T05:59:59Z",
              "name": "learn to write code",
              "needs_grading_count": 0,
              "overrides": Array [],
              "peer_reviews": false,
              "points_possible": 20,
              "position": 1,
              "submission_types": Array [
                "online_text_entry",
                "online_url",
                "media_recording",
                "online_upload",
              ],
              "unlock_at": "2017-03-17T06:00:00Z",
              "updated_at": "2017-03-17T19:15:25Z",
            },
            "assignment_id": 123456789,
            "author": Object {
              "avatar_url": "http://www.fillmurray.com/100/100",
              "display_name": "The Donald",
              "id": "1",
              "short_name": "The Donald",
            },
            "can_unpublish": true,
            "delayed_post_at": null,
            "discussion_subentry_count": 2,
            "discussion_type": "side_comment",
            "html_url": "http://mobiledev.instructure.com/courses/1/discussion_topics",
            "id": "1",
            "last_reply_at": "2016-12-11T04:03:17Z",
            "message": "<p>And why?</p>",
            "only_graders_can_rate": false,
            "permissions": Object {
              "attach": true,
              "delete": true,
              "reply": true,
              "update": true,
            },
            "pinned": false,
            "position": 1,
            "posted_at": "2013-11-14T16:55:00-07:00",
            "published": true,
            "read_state": "read",
            "require_initial_post": false,
            "sort_by_rating": false,
            "subscribed": true,
            "title": "Untitled",
            "unlock_at": null,
            "unread_count": 1,
            "user_can_see_posts": true,
            "user_name": "Ivy Iversen",
          },
          Object {
            "allow_rating": false,
            "assignment": Object {
              "assignment_group_id": 1376,
              "automatic_peer_reviews": false,
              "course_id": 987654321,
              "created_at": "2017-03-17T19:15:25Z",
              "description": "a course that can take any beginner to professional in a matter of microseconds.  Harness the power of your favorite languages like PHP and Cold Fusion.",
              "due_at": "2037-06-01T05:59:00Z",
              "grading_type": "points",
              "html_url": "http://hostname.tld/whatever",
              "id": 123456789,
              "lock_at": "2037-06-01T05:59:59Z",
              "name": "learn to write code",
              "needs_grading_count": 0,
              "overrides": Array [],
              "peer_reviews": false,
              "points_possible": 20,
              "position": 1,
              "submission_types": Array [
                "online_text_entry",
                "online_url",
                "media_recording",
                "online_upload",
              ],
              "unlock_at": "2017-03-17T06:00:00Z",
              "updated_at": "2017-03-17T19:15:25Z",
            },
            "assignment_id": 123456789,
            "author": Object {
              "avatar_url": "http://www.fillmurray.com/100/100",
              "display_name": "The Donald",
              "id": "1",
              "short_name": "The Donald",
            },
            "can_unpublish": true,
            "delayed_post_at": null,
            "discussion_subentry_count": 2,
            "discussion_type": "side_comment",
            "html_url": "http://mobiledev.instructure.com/courses/1/discussion_topics",
            "id": "2",
            "last_reply_at": "2016-12-11T04:03:17Z",
            "message": "<p>And why?</p>",
            "only_graders_can_rate": false,
            "permissions": Object {
              "attach": true,
              "delete": true,
              "reply": true,
              "update": true,
            },
            "pinned": false,
            "position": 1,
            "posted_at": "2013-10-28T14:16:00-07:00",
            "published": true,
            "read_state": "read",
            "require_initial_post": false,
            "sort_by_rating": false,
            "subscribed": true,
            "title": "Getting Started in Biology 101",
            "unlock_at": null,
            "unread_count": 1,
            "user_can_see_posts": true,
            "user_name": "Ivy Iversen",
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onRefresh={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      onViewableItemsChanged={undefined}
      refreshControl={
        <RefreshControlMock
          onRefresh={[Function]}
          progressViewOffset={undefined}
          refreshing={false}
        />
      }
      refreshing={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      testID="announcements.list.list"
      updateCellsBatchingPeriod={50}
      windowSize={21}
    >
      <RCTRefreshControl />
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <TouchableHighlight
            accessibilityLabel={undefined}
            accessibilityTraits={
              Array [
                "button",
              ]
            }
            onPress={[Function]}
            style={
              Array [
                Object {
                  "height": undefined,
                },
                undefined,
                Object {
                  "borderBottomColor": "#C7CDD1",
                  "borderBottomWidth": 0.5,
                },
              ]
            }
            testID="announcements.list.announcement.row-0"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "flex": 1,
                    "flexDirection": "row",
                    "paddingBottom": 12.8,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "paddingTop": 12.8,
                  },
                  Object {
                    "backgroundColor": "white",
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "column",
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  numberOfLines={0}
                  style={
                    Array [
                      Object {
                        "fontFamily": ".SFUIDisplay",
                      },
                      Object {
                        "color": "#2D3B45",
                        "fontSize": 16,
                      },
                      Array [
                        Object {
                          "fontWeight": "600",
                        },
                        undefined,
                      ],
                      Object {
                        "fontFamily": ".SFUIDisplay-semibold",
                      },
                    ]
                  }
                >
                  Untitled
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "fontFamily": ".SFUIDisplay",
                      },
                      Object {
                        "color": "#2D3B45",
                        "fontSize": 16,
                      },
                      Object {
                        "color": "#8B969E",
                        "fontSize": 14,
                        "marginTop": 2,
                      },
                      Object {
                        "fontFamily": ".SFUIDisplay",
                      },
                    ]
                  }
                  testID="announcements.list.announcement.row-0-subtitle-lbl"
                >
                  Nov 14 at 4:55 PM
                </Text>
              </View>
              <View
                style={
                  Object {
                    "flex": 0,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flex": 1,
                      "flexDirection": "row",
                      "justifyContent": "flex-end",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignSelf": "center",
                        "backgroundColor": "transparent",
                        "borderColor": "#c7c7cc",
                        "borderRightWidth": 2,
                        "borderTopWidth": 2,
                        "height": 10,
                        "marginLeft": 7,
                        "transform": Array [
                          Object {
                            "rotate": "45deg",
                          },
                        ],
                        "width": 10,
                      }
                    }
                  />
                </View>
              </View>
            </View>
          </TouchableHighlight>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <TouchableHighlight
            accessibilityLabel={undefined}
            accessibilityTraits={
              Array [
                "button",
              ]
            }
            onPress={[Function]}
            style={
              Array [
                Object {
                  "height": undefined,
                },
                undefined,
                Object {
                  "borderBottomColor": "#C7CDD1",
                  "borderBottomWidth": 0.5,
                },
              ]
            }
            testID="announcements.list.announcement.row-1"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "flex": 1,
                    "flexDirection": "row",
                    "paddingBottom": 12.8,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "paddingTop": 12.8,
                  },
                  Object {
                    "backgroundColor": "white",
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "column",
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  numberOfLines={0}
                  style={
                    Array [
                      Object {
                        "fontFamily": ".SFUIDisplay",
                      },
                      Object {
                        "color": "#2D3B45",
                        "fontSize": 16,
                      },
                      Array [
                        Object {
                          "fontWeight": "600",
                        },
                        undefined,
                      ],
                      Object {
                        "fontFamily": ".SFUIDisplay-semibold",
                      },
                    ]
                  }
                >
                  Getting Started in Biology 101
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "fontFamily": ".SFUIDisplay",
                      },
                      Object {
                        "color": "#2D3B45",
                        "fontSize": 16,
                      },
                      Object {
                        "color": "#8B969E",
                        "fontSize": 14,
                        "marginTop": 2,
                      },
                      Object {
                        "fontFamily": ".SFUIDisplay",
                      },
                    ]
                  }
                  testID="announcements.list.announcement.row-1-subtitle-lbl"
                >
                  Oct 28 at 3:16 PM
                </Text>
              </View>
              <View
                style={
                  Object {
                    "flex": 0,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flex": 1,
                      "flexDirection": "row",
                      "justifyContent": "flex-end",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignSelf": "center",
                        "backgroundColor": "transparent",
                        "borderColor": "#c7c7cc",
                        "borderRightWidth": 2,
                        "borderTopWidth": 2,
                        "height": 10,
                        "marginLeft": 7,
                        "transform": Array [
                          Object {
                            "rotate": "45deg",
                          },
                        ],
                        "width": 10,
                      }
                    }
                  />
                </View>
              </View>
            </View>
          </TouchableHighlight>
        </View>
      </View>
    </RCTScrollView>
  </View>
</Screen>
`;
